storage/libradb/src/pruner/mod.rs

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module provides `Pruner` which manages a thread pruning old data in the background and is
+//! meant to be triggered by other threads as they commit new data to the DB.
+
+use crate::{
+    schema::{
+        retired_state_record::RetiredStateRecordSchema, state_merkle_node::StateMerkleNodeSchema,
+    },
+    OP_COUNTER,
+};
+use failure::prelude::*;
+use logger::prelude::*;
+use schemadb::{ReadOptions, SchemaBatch, DB};
+use sparse_merkle::RetiredRecordType;
+use std::{
+    sync::{
+        mpsc::{channel, Receiver, Sender},
+        Arc, Mutex,
+    },
+    thread::JoinHandle,
+};
+use types::transaction::Version;
+
+#[cfg(test)]
+use std::thread::sleep;
+#[cfg(test)]
+use std::time::{Duration, Instant};
+
+const COUNTER_NAME: &str = "pruner.least_readable_state_version";
+
+/// The `Pruner` is meant to be part of a `LibraDB` instance and runs in the background to prune old
+/// data.
+///
+/// It creates a worker thread on construction and joins it on destruction. When destructed, it
+/// quits the worker thread eagerly without waiting for all pending work to be done.
+pub(crate) struct Pruner {
+    num_versions_to_keep: u64,
+    worker_thread: Option<JoinHandle<()>>,
+    command_sender: Mutex<Sender<Command>>,
+}
+
+impl Pruner {
+    /// creates a worker thread that waits on a channel for pruning commands.
+    pub fn new(db: Arc<DB>, num_versions_to_keep: u64) -> Self {
+        let (command_sender, command_receiver) = channel();
+
+        let worker_thread =
+            std::thread::spawn(move || Worker::new(db, command_receiver).work_loop());
```

Comment:
Let's use [Builder](https://doc.rust-lang.org/std/thread/struct.Builder.html) so we can name the thread.